lib/log_parser.rb

```diff
@@ -27,4 +29,14 @@ def generate_json
   def count_lines
     IO.readlines(@file).size
   end
+
+  def id_players
+    IO.readlines(@file).each do |line|
+      if line.include? 'ClientUserinfoChanged'
+        client_split = line.split(' n\\').last.split('\\t').first
```

Comment:
seria interessante extrair essa lógica para deixar seu código mais declarativo e mais legível, criando um metodo pra encapsular essa lógica do split
```suggestion
        player = get_player(line)
```

```
private 
  def get_player(line)
    line.split(' n\\').last.split('\\t').first
  end
```